protocol/proto/decoder.go

```diff
@@ -57,10 +53,11 @@ func (d *decoder) Decode(m Unmarshaler) error {
 	if err != nil {
 		return err
 	}
+	d.buffer = growDataBufferIfNeeded(d.buffer, sizeEncodingLength+size, d.bytesPool)
```

Comment:
Hm do you need `sizeEncodingLength+size` bytes for the buffer? You can safely overwrite the buffer you used to decode size so you can start the buffer at [0:] no?

Reply:
I could, I actually don't need to change the decoder, but it feels cleaner to me to have the logic in decoder to match the logic change in encoder.